config/webpack/webpack.config.js

```diff
@@ -88,29 +95,30 @@ const devConfig = (options) => {
       },
     },
     output: {
-      path: path.join(rootPath, 'build'),
+      filename: `${filename}.js`,
+      path: outputPath,
+      publicPath,
     },
     devtool: 'cheap-source-map',
     resolveLoader: {
       modules: [path.resolve(path.join(rootPath, 'node_modules'))],
     },
-    mode: 'development',
     stats: { children: false },
-  });
-};
+  };
 
-const prodConfig = (options) => {
-  const filename = '[name]-[chunkhash]';
-  const prodOptions = Object.assign({}, options, { filename });
+  if (!production) {
+    return devConfig;
+  }
 
-  return merge(devConfig(prodOptions), {
-    output: {
-      path: path.resolve('build'),
-      filename: `${filename}.js`,
-    },
+  return merge.strategy({
+    devtool: 'replace',
+  })(devConfig, {
     mode: 'production',
-    devtool: undefined,
-    plugins: [new CleanPlugin('build', { root: options.rootPath, exclude: ['stats.json'] })],
+    devtool: false,
+    plugins: [
+      // I am hesitant on adding the output path to the clean plugin... I know in rails world they dump into folder with other assets....
```

Comment:
Let's get this resolved before we merge.

Reply:
Looked into it. Precompiling adds assets from rails & node side at the same time result in updated assets within intended dir. I think it is fine to include this.